include/aws/common/math.h

```diff
@@ -15,9 +15,9 @@
 AWS_PUSH_SANE_WARNING_LEVEL
 
 /* The number of bits in a size_t variable */
-#if SIZE_MAX == UINT32_MAX
+#if UINTPTR_MAX == UINT32_MAX
```

Comment:
Tsk, tsk, that other library isn't defining SIZE_MAX to spec:

https://www.open-std.org/jtc1/sc22/wg14/www/docs/n1256.pdf
> Each instance of these macros shall be replaced by a constant expression suitable for use
in #if preprocessing directive

Reply:
But seriously, if SIZE_MAX isn't defined, and so that other library is defining it, what makes us think UINTPTR_MAX is defined?

What environment is that user in?

Reply:
linked the issue. artery library used to always override SIZE_MAX when compiled using C++ with non-preprocessor friendly value.

since we only support c99+ and its been ages for all compilers to add those defines to their header, i would not be too concerned with UINTPTR_MAX missing - if its missing then there are likely myriad of other things we dont support.

Reply:
Why can't this be a cmake feature check rather than in-source adhoc #if?